### PR TITLE
Make CheckDifficultyHybridRule a full validation rule

### PR DESCRIPTION
**Pull request by dangershony**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosTimeMaskRule.cs [hunk 1]
```diff
@@ -39,6 +39,7 @@ public override Task RunAsync(RuleContext context)
             this.Logger.LogTrace("Height of block is {0}, block timestamp is {1}, previous block timestamp is {2}, block version is 0x{3:x}.", chainedHeader.Height, chainedHeader.Header.Time, chainedHeader.Previous?.Header.Time, chainedHeader.Header.Version);
 
             var posRuleContext = context as PosRuleContext;
+            posRuleContext.BlockStake = BlockStake.Load(context.ValidationContext.BlockToValidate);
```
Comment (ferdeen) [inline src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosTimeMaskRule.cs]: Could this ever be null ?

Comment (dangershony) [reply src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosTimeMaskRule.cs]: No because his is a partial validation rule so a block will exist

**Comment by dangershony** [comment]
Some integration tests are failing, but not related to this PR.
I will check it out anyway

**Comment by dangershony** [comment]
Ok this is ready to commit, two tests are failing but this is likely because of the BSQ race condition

**Pull request merged**